src/components/enterprise-user-subsidy/data/hooks.jsx

```diff
@@ -11,39 +11,120 @@ import { LICENSE_STATUS } from './constants';
 import {
   fetchSubscriptionLicensesForUser,
   fetchCustomerAgreementData,
+  requestAutoAppliedLicense,
 } from './service';
 import { features } from '../../../config';
 
-export function useSubscriptionLicenseForUser(enterpriseId) {
+/**
+ * Attempts to fetch any existing licenses associated with the authenticated user and the
+ * specified enterprise customer. Priority is given to activated licenses, then assigned
+ * licenses, then revoked licenses.
+ *
+ * @param {string} enterpriseId The UUID of the enterprise customer
+ * @returns An object representing a user's subscription license or null if no license was found.
+ */
+const fetchExistingUserLicense = async (enterpriseId) => {
+  try {
+    const response = await fetchSubscriptionLicensesForUser(enterpriseId);
+    const { results } = camelCaseObject(response.data);
+    /**
+     * Ordering of these status keys (i.e., activated, assigned, revoked) is important as the first
+     * license found when iterating through each status key in this order will be selected as the
+     * applicable license for use by the rest of the application.
+     *
+     * Example: an activated license will be chosen as the applicable license because activated licenses
+     * come first in ``licensesByStatus`` even if the user also has a revoked license.
+     */
+    const licensesByStatus = {
+      [LICENSE_STATUS.ACTIVATED]: [],
+      [LICENSE_STATUS.ASSIGNED]: [],
+      [LICENSE_STATUS.REVOKED]: [],
+    };
+    results.forEach((item) => {
+      licensesByStatus[item.status].push(item);
+    });
+    const applicableLicense = Object.values(licensesByStatus).reduce((acc, licenses) => {
```

Comment:
could also use Object.values(licensesByStatus).flat()[0] here

Reply:
Oh fancy, much cleaner. Will make this change!